refactor(exceptions): reduce redundancy, simplify message generation

mwtoews opened the pull request:
This PR removes some unnecessary `Exception.__init__` calls when only a simple message is used. But where it is overloaded, use `super()` instead of directly calling the superclass.

The error message generation in MFDataException is simplified, and should generate the same information, but without a trailing `\n`.

spaulins-usgs reviewed (approved):
Most of the changes look good.  I recommend that instead of using "pass" for the empty PlotException class, add a comment describing what the PlotException class is.  Maybe something like, "An exception that occurs while plotting"

mwtoews commented:
I've just checked where `PlotException` is used, and it seems this exception is not needed and should be removed, and replaced with `KeyError` and `ImportError`. I've added a second commit to address these proposed changes.

jdhughes-usgs commented:
@mwtoews could you pull in develop so the windows tests will pass
